### PR TITLE
Adds the ability to log to an SQL database

### PR DESCRIPTION
**Pull request by WhiteXZ**

### CODE REVIEW

File: TShockAPI/Commands.cs [hunk 1]
```diff
@@ -893,23 +893,26 @@ private static void RegisterUser(CommandArgs args)
...
-                if (TShock.Users.GetUserByName(user.Name) == null && user.Name != TSServerPlayer.AccountName) // Cheap way of checking for existance of a user
+				if (TShock.Users.GetUserByName(user.Name) == null && user.Name != TSServerPlayer.AccountName)
+					// Cheap way of checking for existance of a user
 				{
 					args.Player.SendSuccessMessage("Account \"{0}\" has been registered.", user.Name);
 					args.Player.SendSuccessMessage("Your password is {0}.", user.Password);
 					TShock.Users.AddUser(user);
-					Log.ConsoleInfo("{0} registered an account: \"{1}\".", args.Player.Name, user.Name);
+					TShock.CharacterDB.SeedInitialData(TShock.Users.GetUser(user));
+					TShock.Log.ConsoleInfo("{0} registered an account: \"{1}\".", args.Player.Name, user.Name);
```
Comment (nicatronTg) [inline TShockAPI/Commands.cs]: This line is duplicated.

File: TShockAPI/TextLog.cs [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	/// </summary>
+	public class TextLog : ILog, IDisposable
+	{
+		private readonly StreamWriter _logWriter;
+		private readonly LogLevel _logLevel;
+
+		/// <summary>
+		/// Log file name
+		/// </summary>
+		public static string fileName;
```
Comment (nicatronTg) [inline TShockAPI/TextLog.cs]: This should be:

```csharp
public static string fileName { get; private set; }
```

**Comment by nicatronTg** [comment]
After what I've noted has been fixed, looks good to me :+1:.

Do not merge this via PR; I'll rebase it.

File: TShockAPI/Commands.cs (hunk 1 above)
Comment (Olink) [inline TShockAPI/Commands.cs]: Which line?  If this is /register it shouldnt be seeding the database anymore...

Comment (nicatronTg) [inline TShockAPI/Commands.cs]: Yeah, that's a good point. Thanks for pointing that out.

**Pull request closed without merging**